src/main/java/com/minecolonies/entity/ai/citizen/fisherman/EntityAIWorkFisherman.java

```diff
@@ -288,17 +291,18 @@ private boolean hasRodButNotEquipped()
     }
 
     /**
-     * Override this method if you want to keep some items in inventory.
+     * Override this method if you want to keep an amount of items in inventory.
      * When the inventory is full, everything get's dumped into the building chest.
      * But you can use this method to hold some stacks back.
      *
-     * @param stack the stack to decide on
-     * @return true if the stack should remain in inventory
+     * @return a list of objects which should be kept.
      */
-    @Override
-    protected boolean neededForWorker(@Nullable final ItemStack stack)
+    protected Map<ItemStorage, Integer> needXForWorker()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1161)